### PR TITLE
#898 add possibility edit customrobots.txt

### PR DESCRIPTION
**Pull request by dborovsky**

### CODE REVIEW

File: Grand.Web/App_Data/Localization/Upgrade/EN_470_480.nopres.xml [hunk 1]
```diff
@@ -118,4 +118,7 @@
   <LocaleResource Name="Admin.Customers.CustomerRoles.Acl.SaveBeforeEdit">
     <Value>You need to save before you can edit access control list.</Value>
   </LocaleResource>
+  <LocaleResource Name="Admin.System.CustomRobotTxt">
```
Comment (PatrykPorabik) [inline Grand.Web/App_Data/Localization/Upgrade/EN_470_480.nopres.xml:121]: Again wrong resource strings. You've forgot to add <Value> tag, please look at the resource string in the previous line.

**Comment by dborovsky** [comment]
#898

File: Grand.Web/App_Data/Localization/Upgrade/EN_470_480.nopres.xml [hunk 2]
```diff
@@ -118,4 +118,7 @@
...
+    Custom robot.txt
```
Comment (PatrykPorabik) [inline Grand.Web/App_Data/Localization/Upgrade/EN_470_480.nopres.xml]: To maintain consistency I would stay in the robots.txt nomenclature.

File: Grand.Web/Areas/Admin/Views/Common/CustomRobotTxt.cshtml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+                <div class="col-12">
+                    <textarea id="Content" name="Content">@Model.Content</textarea>
+                </div>
+                <script>
+                    $(document).ready(function () {
+                        var editor = CodeMirror.fromTextArea(document.getElementById('Content'), {
+                            lineNumbers: true,
+                            matchBrackets: true,
+                            extraKeys: { "Ctrl-Space": "autocomplete" }
```
Comment (PatrykPorabik) [inline Grand.Web/Areas/Admin/Views/Common/CustomRobotTxt.cshtml:46]: Similarly here. It's useless in that case.

File: Grand.Web/Areas/Admin/Views/Common/CustomRobotTxt.cshtml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+@{
+    //page title
+    ViewBag.Title = T("Admin.System.CustomRobotTxt").Text;
+}
+@{
+    pagebuilder.AddScriptParts(ResourceLocation.Head, "~/content/codemirror/codemirror.js", excludeFromBundle: true);
+    pagebuilder.AddScriptParts(ResourceLocation.Head, "~/content/codemirror/mode/css/css.js", excludeFromBundle: true);
+    pagebuilder.AddScriptParts(ResourceLocation.Head, "~/content/codemirror/hint/show-hint.js", excludeFromBundle: true);
+    pagebuilder.AddScriptParts(ResourceLocation.Head, "~/content/codemirror/hint/css-hint.js", excludeFromBundle: true);
+    pagebuilder.AddScriptParts(ResourceLocation.Head, "~/content/codemirror/edit/matchbrackets.js", excludeFromBundle: true);
```
Comment (PatrykPorabik) [inline Grand.Web/Areas/Admin/Views/Common/CustomRobotTxt.cshtml]: As you correctly noticed, it's not possible to implement hints, so why have you added js and CSS resources for hints? :) It's enough to keep only codemirror.js and codemirror.css file.

File: .gitignore [hunk 5]
```diff
@@ -171,3 +171,4 @@ Grand.Web/App_Data/Download/*
 !Grand.Web/App_Data/Download/Index.htm
 Grand.Web/App_Data/TempUploads/*
 !Grand.Web/App_Data/TempUploads/Index.htm
+Grand.Web/wwwroot/robots.custom.txt
```
Comment (PatrykPorabik) [inline .gitignore:174]: Out of the box I would remove it.

File: Grand.Web/App_Data/Localization/defaultResources.grandres.xml [hunk 6]
```diff
@@ -21400,4 +21400,7 @@
   <LocaleResource Name="Admin.Orders.List.OrderTagId">
     <Value>Order's tag</Value>
   </LocaleResource>
+  <LocaleResource Name="Admin.System.CustomRobotTxt">
+    Custom robot.txt
```
Comment (PatrykPorabik) [inline Grand.Web/App_Data/Localization/defaultResources.grandres.xml]: Again you've missed the <value> tag.

File: Grand.Web/App_Data/sitemap.config [hunk 7]
```diff
@@ -156,6 +156,7 @@
         <siteMapNode SystemName="Query editor" grandResource="Admin.System.QueryEditor" PermissionNames="ManageMaintenance" controller="Common" action="QueryEditor" IconClass="fa fa-dot-circle-o"/>
         <siteMapNode SystemName="Custom css" grandResource="Admin.System.CustomCss" PermissionNames="ManageMaintenance" controller="Common" action="CustomCss" IconClass="fa fa-dot-circle-o"/>
         <siteMapNode SystemName="Custom JS" grandResource="Admin.System.CustomJs" PermissionNames="ManageMaintenance" controller="Common" action="CustomJs" IconClass="fa fa-dot-circle-o"/>
+        <siteMapNode SystemName="Custom robot.txt" grandResource="Admin.System.CustomRobotTxt" PermissionNames="ManageMaintenance" controller="Common" action="CustomRobotTxt" IconClass="fa fa-dot-circle-o"/>
```
Comment (PatrykPorabik) [inline Grand.Web/App_Data/sitemap.config]: As previously, please keep the consistency in the nomenclature.

**Review by PatrykPorabik — CHANGES_REQUESTED** [review]
@dborovsky please look at my comments.

**Pull request merged**